### ISSUE TITLE
run `flutter emulators` get crash

### ISSUE THREAD

**Issue by iiunknown** [flutter/flutter#22247]
## command

flutter emulators

## exception

NoSuchMethodError: NoSuchMethodError: The method '[]' was called on null.
Receiver: null
Tried calling: []("avd.ini.displayname")

```
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)
#1      AndroidEmulator.label (package:flutter_tools/src/android/android_emulator.dart:41:34)
#2      Emulator.descriptions (package:flutter_tools/src/emulator.dart:250:18)
#3      Emulator.printEmulators (package:flutter_tools/src/emulator.dart:275:5)
#4      EmulatorsCommand._printEmulatorList (package:flutter_tools/src/commands/emulators.dart:115:14)
#5      EmulatorsCommand._listEmulators (package:flutter_tools/src/commands/emulators.dart:106:7)
<asynchronous suspension>
#6      EmulatorsCommand.runCommand (package:flutter_tools/src/commands/emulators.dart:55:13)
<asynchronous suspension>
#7      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:372:18)
#8      _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#9      _rootRunUnary (dart:async/zone.dart:1132:38)
#10     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#11     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#12     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#13     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#14     Future._complete (dart:async/future_impl.dart:476:7)
#15     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#16     _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async/runtime/libasync_patch.dart:33:20)
#17     _rootRun (dart:async/zone.dart:1124:13)
#18     _CustomZone.run (dart:async/zone.dart:1021:19)
#19     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
#20     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#21     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#22     _runPendingImmediateCallback (dart:isolate/runtime/libisolate_patch.dart:115:13)
#23     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:172:5)
```

## flutter doctor

```
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.13.6 17G65, locale zh-Hans-CN)
    • Flutter version 0.8.2 at /Users/YangLiu/flutter
    • Framework revision 5ab9e70727 (2 weeks ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.2)
    • Android SDK at /Users/YangLiu/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[!] IntelliJ IDEA Ultimate Edition (version 2018.1.6)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.18.0

[!] Connected devices
    ! No devices available

! Doctor found issues in 2 categories.
```

**Comment by zoechi** [comment]
@DanTup is this related to a feature you worked on a while ago?

**Comment by DanTup** [comment]
Indeed. Looks like it may be a bug here:

https://github.com/flutter/flutter/blob/d927c9331005f81157fa39dff7b5dab415ad330b/packages/flutter_tools/lib/src/android/android_emulator.dart#L34-L41

The `label` property is accessing the map directly whereas everything else goes through `_prop()` (which presumably crashes if we couldn't find/parse an ini file for the AVD). I'm out of the office this week so might not get around to fixing until next week but if someone wants to have a go at a PR it should be relatively simple (only additional thing to calling that method is to review the calling code and ensure it doesn't assume a non-null value for `label`).

**Comment by DanTup** [comment]
@thphuc has a fix waiting to be merged in #22503 (I'll merge it once the main build is green).
